agent/src/beerocks/slave/agent_ucc_listener.cpp

```diff
@@ -186,8 +199,50 @@ bool agent_ucc_listener::handle_start_wps_registration(
         err_string = "invalid WpsConfigMethod, supporting only PBC";
         return false;
     }
-    // TODO trigger WPS PBC on the selected band
-    return true;
+
+    auto freq      = band_to_freq(band->second);
+    auto radio_mac = m_backhaul_manager_ctx.freq_to_radio_mac(freq);
+    if (radio_mac.empty()) {
+        err_string = "Failed to get radio mac for " + band->second;
```

Comment:
(nitpick) the CAPI errors should be relevant to the outside world. In this case:
```suggestion
        err_string = "Failed to get radio for " + band->second;
```